sdcm/utils/decorators.py

```diff
@@ -285,3 +285,18 @@ def static_init(cls):
     if hasattr(cls, "static_init"):
         cls.static_init()
     return cls
+
+
+def skip_on_capacity_issues(func: callable) -> callable:
+    """
+    Decorator to skip nemesis that fail due to capacity issues
+    """
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except ClientError as ex:
+            if "InsufficientInstanceCapacity" in str(ex):
+                raise UnsupportedNemesis("Capacity Issue") from ex
```

Comment:
I wonder if we want this behavior in performance tests, skipping the nemesis would change the behavior, and we can get empty report that can skew the results, maybe we better off with failing the nemesis ? 

so maybe introducing a kill switch for this feature ?, so we can disable it for perf tests ?

Reply:
in perf tests we `use_capacity_reservation: true` where it won't start the test without capacity, possibly need to add to more tests (we use it only when placement groups are true too).

Reply:
It can still happen if we miss calculating, or if one wood alters the test, for example different instance types on grow ?

Reply:
different instance types on grow & capacity reservation: https://github.com/scylladb/scylla-cluster-tests/issues/8514

And if one alters the test on the runtime... I don't think we're going to do that in perf tests.

Reply:
so this change is good to go from you POV ?